contracts/core/OTCWrapper.sol

```diff
@@ -732,6 +756,54 @@ contract OTCWrapper is Initializable, OwnableUpgradeable, ReentrancyGuardUpgrade
         emit VaultSettled(_orderID);
     }
 
+    /**
+     * @notice allows user to redeem after expiry
+     * @dev can only be called by the user who is the order buyer
+     * @param _orderID order id
+     */
+    function redeem(uint256 _orderID) external nonReentrant {
+        require(orderStatus[_orderID] == OrderStatus.Succeeded, "OTCWrapper: inexistent or unsuccessful order");
+
+        Order memory order = orders[_orderID];
+
+        require(order.buyer == _msgSender(), "OTCWrapper: sender is not the order buyer");
+
+        // check if vault has enough collateral to cover payout
+        (UtilsWrapperInterface.Vault memory vault, uint256 typeVault, ) = controller.getVaultWithDetails(
+            address(this),
+            order.vaultID
+        );
+
+        (, bool isValidVault) = calculator.getExcessCollateral(vault, typeVault);
+
+        // require that vault is valid (has excess collateral) before redeeming
+        // to avoid allowing redeeming undercollateralized vaults
+        require(isValidVault, "OTCWrapper: insuficient collateral to redeem");
+
+        // transfer otokens to redeem - an approve() by the _msgSender() is required beforehand
+        IERC20(order.oToken).safeTransferFrom(_msgSender(), address(this), order.size);
```

Comment:
this requires an approval before redeeming which is bad UX. in controller can we change it so that [burnOToken](https://github.com/ribbon-finance/GammaProtocol-OTC/blob/f5863ac553444241850cf21ac083035bc808abab/contracts/core/Controller.sol#L945) takes in _args.receiver rather than msg.sender

Reply:
https://github.com/ribbon-finance/GammaProtocol-OTC/pull/5/commits/f6cbe03b97ce5f17ea7ea2625d3f69efff324160